esp-wifi/src/esp_now/mod.rs

```diff
@@ -478,6 +477,46 @@ impl<'d> EspNowReceiver<'d> {
     }
 }
 
+/// The reference counter for properly deinit espnow after all parts are dropped.
+struct EspNowRc<'d> {
+    rc: &'static AtomicU8,
+    inner: PhantomData<EspNow<'d>>
+}
+
+impl<'d> EspNowRc<'d> {
+    fn new() -> Self {
+        static ESP_NOW_RC: AtomicU8 = AtomicU8::new(0);
+        assert_eq!(ESP_NOW_RC.load(Ordering::Acquire), 0);
```

Comment:
Could you please return an error instead?

Reply:
Actually I think this line should never fail since this function can only be called again after all `EspNowRc`s get dropped and `ESP_NOW_RC` will be 0 then.

Reply:
There's nothing stopping me from calling `EspNow::new_*` twice. As the constructors already return an error, I think it would be a bit more polite to not immediately crash if the code doesn't have to.

One more thing: the load-assert-store is race-y as two threads may end up both storing 1. You should either use a `fetch_add` or a critical section.

Reply:
Thanks for your advice. I'll rethink my design here.

Reply:
I've updated code according to your suggestion.